etp-backend/src/test/clj/solita/etp/api/palveluvayla_test.clj

```diff
@@ -0,0 +1,150 @@
+                                         (mock/header "Accept-Language" "fi;q=0.5, sv")))
+                body (-> response :body io/input-stream)
+                correct-pdf (file/find-file ts/*aws-s3-client* (format "energiatodistukset/energiatodistus-%s-%s" todistus-2018-multilingual-id "sv"))]
+            (t/is (= 200 (:status response)))
+            (t/is (IOUtils/contentEquals body correct-pdf))))
+        (t/testing "when prioritising Finnish pdf"
+          (let [response (ts/handler (-> (mock/request :get (str palveluvayla-basepath (format "/pdf/%s" todistus-2018-multilingual-id)))
+                                         (mock/header "Accept-Language" "fi, sv;q=0.5")))
+                body (-> response :body io/input-stream)
+                correct-pdf (file/find-file ts/*aws-s3-client* (format "energiatodistukset/energiatodistus-%s-%s" todistus-2018-multilingual-id "fi"))]
+            (t/is (= 200 (:status response)))
+            (t/is (IOUtils/contentEquals body correct-pdf)))))
+      (t/testing "Asking for a multilingual pdf without Accept-Language returns Finnish pdf"
+        (let [response (ts/handler (mock/request :get (str palveluvayla-basepath (format "/pdf/%s" todistus-2018-multilingual-id))))
+              body (-> response :body io/input-stream)
+              correct-pdf (file/find-file ts/*aws-s3-client* (format "energiatodistukset/energiatodistus-%s-%s" todistus-2018-multilingual-id "fi"))]
+          (t/is (= 200 (:status response)))
+          (t/is (IOUtils/contentEquals body correct-pdf))))
+      (t/testing "Asking for a Swedish pdf without Accept-Language returns it"
+        (let [response (ts/handler (mock/request :get (str palveluvayla-basepath (format "/pdf/%s" todistus-2018-sv-id))))
+              body (-> response :body io/input-stream)
+              correct-pdf (file/find-file ts/*aws-s3-client* (format "energiatodistukset/energiatodistus-%s-%s" todistus-2018-sv-id "sv"))]
+          (t/is (= 200 (:status response)))
+          (t/is (IOUtils/contentEquals body correct-pdf))))
+      (t/testing "Asking for pdf in non-existing language returns 404"
+        (let [response (ts/handler (-> (mock/request :get (str palveluvayla-basepath (format "/pdf/%s" todistus-2018-fi-id)))
+                                       (mock/header "Accept-Language" "sv")))]
+          (t/is (= 404 (:status response))))))
+    (t/testing "Searching for basic information by rakennustunnus returns all versions"
+      (let [response (ts/handler (-> (mock/request :get (str palveluvayla-basepath (format "/json/any?rakennustunnus=%s" rakennustunnus-1)))))
+            body (j/read-value (:body response) j/keyword-keys-object-mapper)]
+        (t/is (= 200 (:status response)))
+        (t/is (= 6 (count body)))
+        (t/is (= #{todistus-2013-fi-id todistus-2018-fi-id todistus-2013-sv-id todistus-2018-sv-id todistus-2013-multilingual-id todistus-2018-multilingual-id} (set (map :id body))))
```

Comment:
Laittaisin näin monen alkion setissä jokaisen omalle rivilleen kuten myös tuon mihin sitä testissä vertaillaan. Erottuu sitten paremmin myös se mitä tähän vertaillaan.